sim-cli/src/main.rs

```diff
@@ -199,3 +218,55 @@ async fn main() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+async fn read_sim_path(data_dir: PathBuf, sim_file: PathBuf) -> anyhow::Result<PathBuf> {
+    let sim_path = if sim_file.is_relative() {
+        data_dir.join(sim_file)
+    } else {
+        sim_file
+    };
+
+    if sim_path.exists() {
+        Ok(sim_path)
+    } else {
+        log::info!("Simulation file '{}' does not exist.", sim_path.display());
+        select_sim_file(data_dir).await
+    }
+}
+
+async fn select_sim_file(data_dir: PathBuf) -> anyhow::Result<PathBuf> {
+    let sim_files = std::fs::read_dir(data_dir.clone())?
+        .filter_map(|f| {
+            f.ok().and_then(|f| {
+                if f.path().extension()?.to_str()? == "json" {
+                    f.file_name().into_string().ok()
+                } else {
+                    None
+                }
+            })
+        })
+        .collect::<Vec<_>>();
+
+    if sim_files.is_empty() {
+        anyhow::bail!(
+            "no simulation files found in {}",
```

Comment:
nit: Now that you've canonicalized this, it should end with a `./` for consistency with the rest of the errors.